### PR TITLE
Correctly dump permissions for a Babelfish logical database

### PR DESCRIPTION
**Pull request by rishabhtanwar29**
### Description
Previously, we were not dumping sysadmin membership for a database
user during database level dump due to which users other than sysadmin
were unable to drop the restored database.

This commit fixes the issue by correctly dumping role memberships for all
the database roles. Additionally, refactored the code a little bit in dump_babel_utils.c

Task: BABEL-4516
Signed-off-by: Rishabh Tanwar <ritanwar@amazon.com>
 
### Check List

- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the PostgreSQL license, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/bin/pg_dump/dumpall_babel_utils.c [hunk 1]
```diff
@@ -304,14 +308,14 @@ getBabelfishRoleMembershipQuery(PGconn *conn, PQExpBuffer buf,
...
-					  "WHERE NOT (ur.rolname ~ '^pg_' AND um.rolname ~ '^pg_')"
-					  "ORDER BY 1,2,3", role_catalog, role_catalog);
+	appendPQExpBufferStr(buf, "SELECT ur.rolname AS roleid, "
+						 "um.rolname AS member, "
+						 "a.admin_option, "
+						 "ug.rolname AS grantor "
+						 "FROM pg_auth_members a "
+						 "INNER JOIN bbf_roles ur on ur.oid = a.roleid "
+						 "INNER JOIN bbf_roles um on um.oid = a.member "
+						 "LEFT JOIN bbf_roles ug on ug.oid = a.grantor "
```
Comment (HarshLunagariya) [inline src/bin/pg_dump/dumpall_babel_utils.c:318]: Why this change? Earlier query was also sufficient, isn't it?

**Review by HarshLunagariya — APPROVED** [review]

File: src/bin/pg_dump/dump_babel_utils.c [hunk 2]
```diff
@@ -1019,26 +1019,27 @@ setBabelfishDependenciesForLogicalDatabaseDump(Archive *fout)
...
-		 * no need to dump catalog entry for it.
+		 * Dump database catalog entry for specified logical database unless
+		 * it's a builtin database (dbid 1, 2, 3 or 4), in which case the db
+		 * will already be present in the target server so no need to dump
+		 * catalog entry for it.
 		 */
-		if (is_builtin_db)
-			appendPQExpBufferStr(buf, " LIMIT 0");
+		appendPQExpBuffer(buf, " FROM ONLY %s a WHERE a.dbid = %d AND a.dbid > 4",
+						  fmtQualifiedDumpable(tbinfo), bbf_db_id);
```
Comment (HarshLunagariya) [inline src/bin/pg_dump/dump_babel_utils.c:1033]: Shouldn't it be > 3 ?

Comment (rishabhtanwar29) [reply src/bin/pg_dump/dump_babel_utils.c:1033]: We have reserved 1 to 4 actually. refer: https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_3_X_DEV/contrib/babelfishpg_tsql/src/dbcmds.c#L308

File: src/bin/pg_dump/dumpall_babel_utils.c (hunk 1 above)
Comment (rishabhtanwar29) [reply src/bin/pg_dump/dumpall_babel_utils.c:318]: Now that we have included role sysadmin in bbf_roles, the `LEFT` join was generating GRANTs for the roles of other logical databases present in the server so need to change the join to more precise `INNER` join.

**Review by HarshLunagariya — APPROVED** [review]

**Review by surendravishnoi — APPROVED** [review]
